src/command-handler.ts

```diff
@@ -31,6 +41,31 @@ class CommandHandler {
         }
     }
 
+    static isAdmin(interaction: Interaction) {
+        return interaction.memberPermissions?.has(PermissionFlagsBits.Administrator)
+            || state.isAdmin(interaction.user.id);
+    }
+
+    static assertIsPrivilegedRole(interaction: Interaction) {
+        if (!interaction.guild) {
+            throw new Error(INVALID_TEXT_CHANNEL);
+        }
+        // If there is no saved privileged role then reject the command
+        if (!state.hasPrivilegedRole(interaction.guild.id)) {
```

Comment:
Should we really fail commands if there is no role set? Why not just treat this case as "admins only". If the role is set, then "admins and <role> only"

Reply:
This is only invoked if the interacting user is not already an admin